sw/device/silicon_creator/lib/ownership/owner_block.c

```diff
@@ -256,11 +256,15 @@ rom_error_t owner_block_info_apply(const owner_flash_info_config_t *info) {
 }
 
 rom_error_t owner_keyring_find_key(const owner_application_keyring_t *keyring,
-                                   uint32_t key_alg, uint32_t key_id,
-                                   size_t *index) {
+                                   uint32_t key_id, size_t *index) {
   for (size_t i = 0; i < keyring->length; ++i) {
-    if (keyring->key[i]->key_alg == key_alg &&
-        keyring->key[i]->data.id == key_id) {
+    uint32_t id = keyring->key[i]->data.id;
+    if ((keyring->key[i]->key_alg & kOwnershipKeyAlgCategoryMask) ==
+        kOwnershipKeyAlgCategoryHybrid) {
+      // The ID of a hybrid key is the sum of the IDs of each key.
```

Comment:
How do we know this won't overflow/that each hybrid key ID will be unique compared to other hybrid or non-hybrid keys?

Reply:
We don't know that (any more than we know that the least significant word of any key will be unique).  I can change this operation to XOR instead of a sum to not lose the potential carry bit.  IMHO, we'll still have the same potential uniqueness problem.

Another alternative is to use a 64-bit integer and concatenate the two u32's together as the key ID.

WDYT?

Reply:
Another option is to have the key look up operations implemented in a similar way to the ROM, and then add a separate function to verify the hybrid policy, to ensure that both keys are part of the same key slot. 

```
owner_keyring_hybrid_validate(ecdsa_key_id, spx_id)
```

I was thinking that this way we can avoid adding the new switch case statements added to the rom_ext to simplify the hardening.

Reply:
I decided to xor the `key_ids`.  This allows the possibility of having an SPX-only signature in the future.